PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/tree/ViewNode.java

```diff
@@ -351,15 +350,87 @@ public void highlight(Pane treePane) {
   }
 
   /**
-   * Remove the highlight of the area of this node.
+   * Remove the highlightNode of the area of this node.
    *
-   * @param treePane the pane from which to remove the highlight area.
+   * @param treePane the pane from which to remove the highlightNode area.
    */
   public void removeHighlight(Pane treePane) {
     treePane.getChildren().remove(highlightArea);
     highlightArea = null;
   }
 
+  /**
+   * Highlight the given node and its child nodes. First search for matches in any of the parent
+   * nodes, then search for matches in the child nodes.
+   *
+   * @param node  the node.
+   * @param color the color.
+   */
+  public void highlightNode(IPhylogeneticTreeNode node, Color color) {
+    IPhylogeneticTreeNode parent = dataNode;
+    while (parent.hasParent()) {
+      parent = parent.getParent();
+      if (parent.equals(node)) {
+        setRecursiveColor(color);
+        return;
+      }
+    }
+    highlightChildNode(node, color);
+  }
```

Comment:
This method iterates from a given node upwards, and then from that (grand)parent calls a method that recurses down. Seems a bit unnecessary?

Can you not set the color in the `while` loop?